tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,527 @@
+
+    Args:
+        dataset: An existing TUC dataset
+        attributes: list of attribute names to be added to dataset
+        attribute_types: dictionary for non-default types, attribute name is the key and
+            AttributeType is the value
+        attribute_descriptions: dictionary for attribute descriptions, attribute name is the
+            key and the attribute description is the value
+
+    Returns:
+        Updated Dataset
+
+    Raises:
+        requests.HTTPError: If any HTTP error is encountered
+        TypeError: If the attributes argument is not an Iterable
+        ValueError: If the dataset is a unified dataset
+        ValueError: If an attribute passed in already exists in the dataset
+
+    """
+    dataset_name = dataset.name
+    if dataset.upstream_datasets():
+        raise ValueError(f"{dataset_name} is not a source dataset")
+
+    # Check input type is correct
+    if not isinstance(attributes, Iterable):
+        raise TypeError("attributes arg must be an Iterable")
+
+    # Get current dataset attributes
+    target_dataset_attributes = dataset.attributes
+    existing_attributes = []
+    for attr in target_dataset_attributes.stream():
+        existing_attributes.append(attr.name)
+
+    # Check that none of the new attribute names already exist
+    for attribute_name in attributes:
+        if attribute_name in existing_attributes:
+            # This attribute already exists
+            raise ValueError(
+                f"An attribute with name '{attribute_name}' already exists in {dataset_name}"
+            )
+
+    # Add attributes to dataset
+    for attribute_name in attributes:
+        attr_spec_dict = _make_spec_dict(
+            attribute_name=attribute_name,
+            attribute_types=attribute_types,
+            attribute_descriptions=attribute_descriptions,
+        )
+        target_dataset_attributes.create(attr_spec_dict)
+        LOGGER.info(f"Created attribute '{attribute_name}' in {dataset_name}")
+
+    return dataset
+
+
+def edit_attributes(
+    *,
+    dataset: Dataset,
+    attribute_types: Optional[Dict[str, attribute_type.AttributeType]] = None,
+    attribute_descriptions: Optional[Dict[str, str]] = None,
+    override_existing_types: bool = True,
+) -> Dataset:
+    """
+    Edits existing attributes in a dataset. The attribute type and/or descriptions can be updated
+    to new values. Attributes that the user wants to update must either in either the
+    attribute_types or attribute_descriptions dictionaries or both. The default attribute type
+    will be ARRAY STRING. To set non-default attribute types, you must define them in the
+    attribute_types dictionary. Any attribute descriptions can be specified in the
+    attribute_descriptions dictionary. If only the attribute_descriptions dictionary is defined,
+    it will be assumed the attribute types should be set to the default (ARRAY STRING).
```

Comment:
nit: format and consistent grammar plus one type
```suggestion
    """Edit existing attributes in a dataset
    
    The attribute type and/or descriptions can be updated to new values. Attributes that will be
    updated must be in either the attribute_types or attribute_descriptions dictionaries or
    both. The default attribute type will be ARRAY STRING. To set non-default attribute types, they
    must be defined in the attribute_types dictionary. Any attribute descriptions can be specified
    in the attribute_descriptions dictionary. If only the attribute_descriptions dictionary is
    defined, their attribute types will be set to the default ARRAY STRING.
```